src/modules/sync/utils/contentTypeHelpers.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  ContentTypeContracts,
+  ContentTypeElements,
+  ContentTypeSnippetContracts,
+  ElementContracts,
+} from "@kontent-ai/management-sdk";
+import {
+  DomHtmlNode,
+  DomNode,
+  nodeParse,
+  ParseResult,
+  ResolveDomHtmlNodeType,
+  ResolveDomTextNodeType,
+  transformToJson,
+} from "@kontent-ai/rich-text-resolver";
+
+const isAssetElement = (
+  element: ElementContracts.IContentTypeElementContract,
+): element is ContentTypeElements.IAssetElement => element.type === "asset";
+
+const isLinkedItemElement = (
+  element: ElementContracts.IContentTypeElementContract,
+): element is ContentTypeElements.ILinkedItemsElement => element.type === "modular_content";
+
+const resolveAssetIdsDomHtmlNode: ResolveDomHtmlNodeType = (node, traverse) => {
+  switch (node.tagName) {
+    case "figure":
+      return node.attributes["data-asset-id"];
+    case "a":
+      return node.attributes["data-asset-id"];
+    default: {
+      if (node.children.length) {
+        return node.children.map(c => traverse(c));
+      }
+      return null;
+    }
+  }
+};
+
+const resolveItemIdsDomHtmlNode: ResolveDomHtmlNodeType = (node, traverse) => {
+  switch (node.tagName) {
+    case "a":
+      return node.attributes["data-item-id"];
+    default: {
+      if (node.children.length) {
```

Comment:
Is it necessary to return `null` when there are no children? I believe it would also work to return an empty array. I you'd return an empty array instead, you would not need this `if`.

Reply:
The same applies to the `resolveAssetIdsDomHtmlNode` above. 😉